app/src/main/java/com/codesoom/assignment/application/AuthenticationService.java

```diff
@@ -0,0 +1,49 @@
+package com.codesoom.assignment.application;
+
+import com.codesoom.assignment.domain.User;
+import com.codesoom.assignment.domain.UserRepository;
+import com.codesoom.assignment.errors.LoginFailException;
+import com.codesoom.assignment.utils.JwtUtil;
+import io.jsonwebtoken.Claims;
+import org.springframework.stereotype.Service;
+
+/**
+ * 사용자의 인증과 관련한 로직을 수행합니다.
+ */
+@Service
+public class AuthenticationService {
+
+    private final JwtUtil jwtUtil;
+    private  final UserRepository userRepository;
+
+    public AuthenticationService(JwtUtil jwtUtil, UserRepository userRepository) {
+        this.jwtUtil = jwtUtil;
+        this.userRepository = userRepository;
+    }
+
+    /**
+     * 사용자 정보로 로그인하여 인증 토큰을 리턴합니다.
+     * @param email 로그인 이메일
+     * @param password 로그인 비밀번호
+     * @return 인증 토큰
+     */
+    public String login(String email, String password) {
+        User user = userRepository.findByEmail(email)
+                .orElseThrow(() -> new LoginFailException(email));
+
+        if (!user.authenticate(password)) {
```

Comment:
이렇게 수정해주면 읽고 이해하기 더 쉽지 않을까요?
```suggestion
        if (!user.isValid(password)) {
```